### PR TITLE
NEW: build scripts

### LINKED ISSUE #15: build scripts
its can be helpful to have some bash scripts to automate build

### PR DESCRIPTION
**Pull request by mirzakhany**
this pull request will provide two build scripts. we can use them in a Makefile also

### CODE REVIEW

**Comment by saeedghx68** [comment]
@mirzakhany  please put FIX, OPT, or NEW in the first of your commit's message.
If you can please squash your commits as one

File: README.md [hunk 1]
```diff
@@ -11,4 +11,47 @@ this project is going to be a template for starting django projects. and will co
 * [ ] ?
 
 
+# How to build docker images.
+ 
+we have two script in `deploy/scripts` directory to make build process easier in ci/cd. but we can also use them durring development.
 
+### first of all set required variables
```
Comment (hramezani) [inline README.md]:
```suggestion
### First of all set required variables
```

File: README.md [hunk 2]
```diff
@@ -11,4 +11,47 @@ this project is going to be a template for starting django projects. and will co
 * [ ] ?
 
 
+# How to build docker images.
+ 
+we have two script in `deploy/scripts` directory to make build process easier in ci/cd. but we can also use them durring development.
```
Comment (hramezani) [inline README.md]:
```suggestion
We have two scripts in `deploy/scripts` directory to make build process easier in ci/cd. but we can also use them during development.
```

File: README.md [hunk 3]
```diff
@@ -11,4 +11,47 @@ this project is going to be a template for starting django projects. and will co
...
+```bash
+export DOCKER_REGISTRY="index.docker.io";
+export DOCKER_USER="";
+export DOCKER_PASS="";
+
+export BASE_IMAGE_REPO="mirzakhani/django_boilerplate_base";
+export IMAGE_REPO="mirzakhani/django_boilerplate";
+```
+
+`DOCKER_REGISTRY` : is to set your docker register address, default address will work for dockerhub.
```
Comment (hramezani) [inline README.md]:
```suggestion
`DOCKER_REGISTRY`: is to set your docker register address, default address will work for dockerhub.
```

File: README.md [hunk 4]
```diff
@@ -11,4 +11,47 @@ this project is going to be a template for starting django projects. and will co
...
+
+`DOCKER_USER` and `DOCKER_PASS` are used to login to your docker reqistery. !!! Again dont forget to put this file in .gitignore !!!
```
Comment (hramezani) [inline README.md]:
```suggestion
`DOCKER_USER` and `DOCKER_PASS` are used to login to your docker registry. !!! Again don't forget to put this file in .gitignore !!!
```

File: deploy/scripts/build_base_image.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+DOCKER_FILE="deploy/BaseDockerfile"
+BASE_URL="https://$DOCKER_REGISTRY/v1/"
+
+REQ_HASH=$(cat requirements.txt $DOCKER_FILE | md5sum | awk '{print $1}')
+
+echo "check docker registry for image tag: $BASE_IMAGE_REPO:$REQ_HASH"
+IMAGE=$(curl -s -u $DOCKER_USER:"$DOCKER_PASS" -X GET $BASE_URL/$BASE_IMAGE_REPO/tags/list | grep $REQ_HASH)
+
+if [ ! $IMAGE ]; then
+    echo "Base image for base not exists, now building ..."
```
Comment (hramezani) [inline deploy/scripts/build_base_image.sh]:
```suggestion
    echo "Base image for base does not exist, now building ..."
```

**Review by hramezani — APPROVED** [review]
:+1:

**Pull request merged**